### PR TITLE
Add PlPredicate, PlModule classes

### PR DESCRIPTION
**Pull request by kamahen**
This is a follow-on to PR https://github.com/SWI-Prolog/packages-cpp/pull/31

### CODE REVIEW

File: SWI-cpp2.h [hunk 1]
```diff
@@ -583,6 +582,34 @@ class PlTerm_recorded : public PlTerm
...
+    : WrappedC<module_t>(PL_new_module(name.C_))
+  { verify();
+  }
+};
+
+class PlPredicate : public WrappedC<predicate_t>
+{
+public:
+  explicit PlPredicate()
+    : WrappedC<predicate_t>(PlPredicate::null) { }
```
Comment (JanWielemaker) [inline SWI-cpp2.h]: I don't think there is any scenario I'm aware of where it is useful to have a NULL predicate pointer.

File: SWI-cpp2.h [hunk 2]
```diff
@@ -583,6 +582,34 @@ class PlTerm_recorded : public PlTerm
   explicit PlTerm_recorded(record_t r) { PlCheck(PL_recorded(r, C_)); }
 };
 
+class PlModule : public WrappedC<module_t>
+{
+public:
```
Comment (JanWielemaker) [inline SWI-cpp2.h:585]: Here we may  have a null constructor.   Practically all API functions accept a NULL module as a shorthand for the current calling context, defaulting to `user` if such a context does not exist (when we make calls to Prolog without being called by Prolog).

File: SWI-cpp2.h (hunk 1 above)
Comment (kamahen) [reply SWI-cpp2.h]: Removed (it was in rocksdb, but I've now made it explicit)

File: SWI-cpp2.h (hunk 2 above)
Comment (kamahen) [reply SWI-cpp2.h:585]: I've added a constructor that takes module_t, so PlModule(PlModule::null) is now possible.

**Comment by JanWielemaker** [comment]
Great.   I've lost track a little with all the PRs.   When done, can you tell me by mail which to merge?

**Comment by JanWielemaker** [comment]
Thanks.  Squashed with #31.  Added an additional patch to reduce the time needed for the overflow test by temporary lowering the stack limit.

**Pull request closed without merging**